Source/Dafny/MergeOrdered.cs

```diff
@@ -16,14 +16,15 @@ namespace Microsoft.Dafny;
 /// </summary>
 public class MergeOrdered<T> : IObservable<T>, IObserver<IObservable<T>> {
   private readonly Queue<IObservable<T>> allUpdates = new();
-  private IDisposable innerSubscription;
+  private bool idle = true;
   private bool outerCompleted;
   private readonly Subject<T> result = new();
 
   public void OnNext(IObservable<T> next) {
     lock (this) {
-      if (innerSubscription == null) {
-        innerSubscription = next.Subscribe(InnerNext, InnerError, InnerCompleted);
```

Comment:
Here `innerSubscription` was assigned which is supposed to be nulled in `InnerCompleted`, but if `next.Subscribe` calls the passed OnComplete handler before it returns, `InnerCompleted` in this case, which happens when `next` is `Observable.Empty<T>()`, then `innerSubscription` would end up never getting nulled again.